vroom/system.vroom

```diff
@@ -362,4 +362,33 @@ maktaba#system#Or to chain commands.
   ! false || echo FAILURE
   $ FAILURE
 
+To execute system calls asynchronously, use CallAsync.
+
+  :let g:syscall = maktaba#syscall#Create(['echo', 'hi'])
+  :function Callback(env, stdout, stderr, exit_code) abort<CR>
+  |  let g:env = a:env<CR>
+  |  let g:callback_stdout = a:stdout<CR>
+  |  let g:callback_exit_code = a:exit_code<CR>
+  |endfunction
+  :call g:syscall.CallAsync('Callback')
+  ! .*echo hi; vim --servername .* --remote-expr "maktaba#syscall#AsyncDone.*
+  :let g:deadline = localtime() + 5 " wait for at most 4 seconds
+  :while !exists('g:callback_exit_code') && localtime() < g:deadline
```

Comment:
As far as I can see, this will fail unless the callback fires before you even have a chance to enter the loop. Try something like this:
```vim
:while !exists('g:callback_exit_code') && localtime() < g:deadline
:endwhile
:call maktaba#ensure#IsTrue(exists('g:callback_exit_code'),
| 'Async callback was expected to complete within 2 seconds')
:echomsg g:callback_stdout
```